lib/index.js

```diff
@@ -393,7 +401,7 @@ Mongoose.prototype.model = function(name, schema, collection, skipInit) {
   }
 
   var connection = options.connection || this.connection;
-  model = this.Model.compile(name, schema, collection, connection, this);
+  model = this.Model.compile(model || name, schema, collection, connection, this);
```

Comment:
When I first looked at this, I thought this bit was still inside the `if (typeof name === 'function')..` scope.